main.py

```
@@ -604,0 +1,1133 @@
+            f"Process completed successfully. Output saved in {validated_format} format."
+        )
+    except Exception as e:
+        logger.error(f"An error occurred during the process: {e}", exc_info=True)
+        raise
+
+def _validate_output_format(output_format: str) -> str:
+    """Validate and normalize output format."""
+    normalized_format = output_format.lower()
+    if normalized_format not in ["markdown", "json", "xml"]:
+        logger.warning(
+            f"Invalid output format: {output_format}. Using markdown instead."
```

Ruff (0.8.2)

Logging statement uses f-string

(G004)